controller/appcontroller.go

```diff
@@ -254,13 +255,27 @@ func (ctrl *ApplicationController) setAppManagedResources(a *appv1.Application,
 }
 
 // returns true of given resources exist in the namespace by default and not managed by the user
-func isKnownOrphanedResourceExclusion(key kube.ResourceKey) bool {
+func isKnownOrphanedResourceExclusion(key kube.ResourceKey, proj *appv1.AppProject) bool {
 	if key.Namespace == "default" && key.Group == "" && key.Kind == kube.ServiceKind && key.Name == "kubernetes" {
 		return true
 	}
 	if key.Group == "" && key.Kind == kube.ServiceAccountKind && key.Name == "default" {
 		return true
 	}
+
+	list := proj.Spec.OrphanedResourceWhitelist
+	for _, item := range list {
```

Comment:
I propose to use the following logic:

* if Kind/Name is not specified then we assume that value is `*` (match everything). 
* group must be specified because Kubernetes has empty group, so we cannot assume that empty value means everything.